src/tests/ftest/dfuse/dfuse.py

```diff
@@ -0,0 +1,113 @@
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+from __future__ import print_function
+
+import os
+import sys
+import json
+from avocado import Test
+
+sys.path.append('./util')
+sys.path.append('../util')
+sys.path.append('../../../utils/py')
+sys.path.append('./../../utils/py')
+
+import AgentUtils
+import server_utils
+import write_host_file
+from daos_api import DaosContext, DaosPool, DaosContainer
+
+class Dfuse(Test):
+    """
+    Runs Dfuse test.
+    """
+
+    def __init__(self, *args, **kwargs):
+
+        super(Dfuse, self).__init__(*args, **kwargs)
+
+        self.hostlist_servers = None
+        self.hostlist_clients = None
+        self.agent_sessions = None
+        self.prefix = None
+
+    def setUp(self):
+
+        # get paths from the build_vars generated by build
+        with open('../../../.build_vars.json') as build_file:
+            build_paths = json.load(build_file)
+        self.prefix = build_paths['PREFIX']
+        basepath = os.path.normpath(os.path.join(self.prefix, '..'))
+
+        server_group = self.params.get("server_group", '/server/',
+                                       'daos_server')
+
+        self.hostlist_servers = self.params.get("test_servers", '/run/hosts/')
+        hostfile_servers = (
+            write_host_file.write_host_file(self.hostlist_servers,
+                                            self.workdir))
+        print("Host file servers is: {}".format(hostfile_servers))
+
+        self.hostlist_clients = self.params.get("test_clients", '/run/hosts/')
+
+        # start server
+        self.agent_sessions = AgentUtils.run_agent(basepath,
+                                                   self.hostlist_servers,
+                                                   self.hostlist_clients)
+        server_utils.run_server(hostfile_servers, server_group, basepath)
+
+    def tearDown(self):
+        if self.agent_sessions:
+            AgentUtils.stop_agent(self.hostlist_clients, self.agent_sessions)
+        server_utils.stop_server(hosts=self.hostlist_servers)
+
+    def test_dfuse(self):
+        """Try and run something over fuse
+
```

Comment:
Required docstring below.  Also every test should have a unique tag so that just that one test can be run manually.  And its also handy to have it be part of a group of tests.  So for a test of delete pool we have a "delete_pool" tag and a "pool" tag and "vm" and "regression".  Then you can run just that test with delete_pool and run it as part of all the pool tests because it also has the pool tag.

Test ID: test case ID goes here
Test Description: description of the test case
Use Cases: list of use cases that are to at least some degree covered by this test case
:avocado: tags=regression,pool,vm

Reply:
Makes sense - although I'm very much still at the stage of being able to get anything to run at all.